### ISSUE TITLE
Pretrained models on other dataset: NuScenes

### ISSUE THREAD

**Issue by ducminhkhoi** [cv-rits/MonoScene#8]
Hi @anhquancao,

Thanks so much for your paper and your implementation. Do you have your pretrained model on the NuScenes? If yes, could you share it? The reason is that I want to build upon your work on the NuScenes dataset but there exists a large domain gap between the two (SemanticKITTI and NuScenes) so the pretrained on SemanticKITTI works does not well on the NuScenes.

Thanks!

**Comment by anhquancao** [comment]
Hi @ducminhkhoi,
Thank you for your interest in our work. 
We didn't train MonoScene on NuScenes since there is no SSC ground truth available. 
However, you can follow the dataset preparation in the following to generate the SSC label for NuScenes.
https://github.com/yanx27/JS3C-Net#data-preparation
We didn't do this due to time constraints.

**Comment by ducminhkhoi** [comment]
Thanks for your reply. 

In your opinion, if I train this model on the NuScenes dataset, what configuration should I need to change?

**Comment by anhquancao** [comment]
To train on NuScenes, you need to change

1. the **vox_origin**  and **scene_size** variables in the following line
https://github.com/cv-rits/MonoScene/blob/master/monoscene/data/semantic_kitti/kitti_dataset.py#L41
It is the origin of the sampled voxel grid in the Velodyne coordinate.
For Kitti dataset, the grid size is (x, y, z)=(51.2m, 51.2m, 6.4m) and the voxel origin is [0, -25.6, -2]. You can check the direction of the x, y, z axes in the link: http://www.cvlibs.net/datasets/kitti/setup.php


2. The **projection parameters** in lines 87 and 88 in the kitti_dataset.py:
https://github.com/cv-rits/MonoScene/blob/master/monoscene/data/semantic_kitti/kitti_dataset.py#L88
- P: 4x4 is the camera intrinsic
- T_velo_2_cam: 4x4: is the transformation matrix map from the Velodyne coordinate to the camera coordinate
They are used in the dataloader to compute the mapping between the centroids of voxels to the 2D pixels.

**Comment by ducminhkhoi** [comment]
Thanks!

**Comment by AgentEXPL** [comment]
> To train on NuScenes, you need to change
> 
> 1. the **vox_origin**  and **scene_size** variables in the following line
>    https://github.com/cv-rits/MonoScene/blob/master/monoscene/data/semantic_kitti/kitti_dataset.py#L41
>    It is the origin of the sampled voxel grid in the Velodyne coordinate.
>    For Kitti dataset, the grid size is (x, y, z)=(51.2m, 51.2m, 6.4m) and the voxel origin is [0, -25.6, -2]. You can check the direction of the x, y, z axes in the link: http://www.cvlibs.net/datasets/kitti/setup.php
>    For NuScenes, the grid size should be (x, y, z)=(51.2m, 51.2m, 6.4m) and the voxel origin is [-25.6, 0, -2]
> 2. The **projection parameters** in lines 87 and 88 in the kitti_dataset.py:
>    https://github.com/cv-rits/MonoScene/blob/master/monoscene/data/semantic_kitti/kitti_dataset.py#L88
> 
> * P: 4x4 is the camera intrinsic
> * T_velo_2_cam: 4x4: is the transformation matrix map from the Velodyne coordinate to the camera coordinate
>   They are used in the dataloader to compute the mapping between the centroids of voxels to the 2D pixels.

Hi, @anhquancao Could you please provide more detailed explanations on the meaning of cam_pose and vox_origin. Is cam_pose representing the pose of camera in world coordinate (room)? Which coordinate does the vox_origin belong to?

**Comment by anhquancao** [comment]
Hi @AgentEXPL, sorry for confusing variable name, cam_pose is the transformation matrix from the reconstructed coordinate to the camera coordinates. In the case of SemanticKITTI, I reconstructed in the lidar coordinate so cam_pose map from lidar coordinate to camera coordinates. 
vox_origin belongs to the reconstructed coordinate, which is lidar coordinate in the case of SemanticKITTI.

**Comment by AgentEXPL** [comment]
@anhquancao Thanks a lot for the reply. One more question, how to compute vox_origin for each picture in NYU dataset? The vox_origin is different for each sample picture. Does the vox_origin's (x,y) value represents the plus of a fixed value and the (x,y) value of the camera pose?

**Comment by anhquancao** [comment]
The vox_origin was provided in the original dataset.
https://github.com/astra-vision/MonoScene/blob/master/monoscene/data/NYU/preprocess.py#L79

**Comment by AgentEXPL** [comment]
> The vox_origin was provided in the original dataset. https://github.com/astra-vision/MonoScene/blob/master/monoscene/data/NYU/preprocess.py#L79

Now I am trying to produce my own dataset (of an indoor robot scenario) in the format of NYU dataset. So I want to know how to compute the value of vox_origin, rather than obtaining it from the provided dataset.

**Comment by anhquancao** [comment]
You can define in the world coordinate a cube with size HxWxD meters, voxelize it, keep only voxels that are projected on the image and up to a depth value that you want (e.g. 10m). Make sure the cube is large enough that it can contain all the scene.
